library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -78,10 +76,21 @@ private void createNotificationChannel() {
         public void onUpdate(MigrationStatus migrationStatus) {
             NotificationInformation notification = notificationCreator.createNotification(migrationStatus);
 
-            if (migrationStatus.status() == Status.COMPLETE || migrationStatus.status() == Status.DB_NOT_PRESENT) {
-                stackNotification(notification);
-            } else {
-                updateNotification(notification);
+            switch (notification.notificationStackState()) {
+                case SINGLE_PERSISTENT_NOTIFICATION:
+                    updateNotification(notification);
+                    break;
+                case STACK_NOTIFICATION_DISMISSIBLE:
+                    stackNotification(notification);
+                    break;
+                case STACK_NOTIFICATION_NOT_DISMISSIBLE:
```

Comment:
Isn't this a condition the client could create?

Reply:
Yes, I've left this for now because in a follow up PR I would like to do the following:

> Introduce a new interface that is used to collate the different types of notifications that the download-manager supports. Attach this new interface to both the DownloadService and the MigrationService.